lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/VortexDriver.java

```diff
@@ -0,0 +1,180 @@
+import java.util.logging.Logger;
+
+/**
+ * REEF Driver for Vortex.
+ */
+@Unit
+public final class VortexDriver {
+  private static final Logger LOG = Logger.getLogger(VortexDriver.class.getName());
+  private final EvaluatorRequestor evaluatorRequestor;
+  private final VortexManager vortexManager;
+  private final CommandSender commandSender;
+
+  private final int evalMem;
+  private final int evalNum;
+  private final int evalCores;
+
+  private final VortexThreadPool vortexThreadPool;
+  private final ExecutorService vortexUserThread = Executors.newSingleThreadExecutor();
+  private final VortexUserCode vortexUserCode;
+
+  @Inject
+  public VortexDriver(final EvaluatorRequestor evaluatorRequestor,
+                      final CommandSender commandSender,
+                      final VortexManager vortexManager,
+                      final VortexUserCode vortexUserCode,
+                      final VortexThreadPool vortexThreadPool,
+                      @Parameter(VortexConfiguration.ExecutorMem.class) final int evalMem,
+                      @Parameter(VortexConfiguration.ExecutorNum.class) final int evalNum,
+                      @Parameter(VortexConfiguration.ExecutorCores.class) final int evalCores) {
+    this.evaluatorRequestor = evaluatorRequestor;
+    this.vortexManager = vortexManager;
+    this.commandSender = commandSender;
+    this.vortexUserCode = vortexUserCode;
+    this.vortexThreadPool = vortexThreadPool;
+    this.evalMem = evalMem;
+    this.evalNum = evalNum;
+    this.evalCores = evalCores;
+  }
+
+  /**
+   * Driver started.
+   */
+  public final class StartHandler implements EventHandler<StartTime> {
+    @Override
+    public void onNext(final StartTime startTime) {
+      // Initial Evaluator Request
+      evaluatorRequestor.submit(EvaluatorRequest.newBuilder()
+          .setNumber(evalNum)
+          .setMemory(evalMem)
+          .setNumberOfCores(evalCores)
+          .build());
+
+      // Run the user code as a separate thread
+      // todo: handle user code exceptions
+      vortexUserThread.execute(new Runnable() {
+        @Override
+        public void run() {
+          vortexUserCode.start(vortexThreadPool);
+          vortexManager.terminate();
+        }
+      });
+    }
+  }
+
+  /**
+   * Container allocated.
+   */
+  public final class AllocatedEvaluatorHandler implements EventHandler<AllocatedEvaluator> {
+    @Override
+    public void onNext(final AllocatedEvaluator allocatedEvaluator) {
+      LOG.log(Level.INFO, "Container allocated");
+
+      final Configuration executorConfiguration = ExecutorConfiguration.CONF
+          .set(ExecutorConfiguration.NUM_OF_THREADS, evalCores) // NUM_OF_THREADS = evalCores
+          .build();
+
+      final Configuration contextConfiguration = ContextConfiguration.CONF
+              .set(ContextConfiguration.IDENTIFIER, allocatedEvaluator.getId() + "_context")
+              .set(ContextConfiguration.ON_MESSAGE, LocalExecutor.class)
```

Comment:
I agree it is a bad name. I'm thinking of renaming it to `VortexWorker`. (And `VortexManager` to `VortexMaster` for symmetricity)